### ISSUE TITLE
[java] MissingSerialVersionUID false-negative with abstract classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1352]
**Affects PMD Version:** any

**Rule:** errorprone / MissingSerialVersionUID

**Description:**
As mentioned in #1350 and described in https://stackoverflow.com/questions/893259/should-an-abstract-class-have-a-serialversionuid abstract classes also should have a serialversionUID field.

Currently we ignore abstract classes which is wrong.

**Code Sample demonstrating the issue:**

```
import java.io.Serializable; 
public abstract class Foo implements Serializable { 
} 
```
